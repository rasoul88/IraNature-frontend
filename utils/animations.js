import { keyframes } from "styled-components";

export const moveInLeft = keyframes`
    0% {
        opacity: 0;
        transform: translateX(-10rem);
    }

    80% {
        transform: translateX(1rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;

export const moveInRight = keyframes`
    0% {
        opacity: 0;
        transform: translateX(10rem);
    }

    80% {
        transform: translateX(-1rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;

export const moveInBottom = keyframes`
    0% {
        opacity: 0;
        transform: translateY(3rem);
    }

    100% {
        opacity: 1;
        transform: translateY(0);
    }
`;

export const Shake = keyframes`
    0% {
        transform: translateX(-.3rem);
    }

    100% {
        transform: translateX(.3rem);
    }
`;

export const pageInRight = keyframes`
    0% {
        opacity: 0;
        transform: translateX(200%);
    }

    80% {
        transform: translateX(-10rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;
