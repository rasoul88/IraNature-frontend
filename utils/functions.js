import { toast } from "react-toastify";
import Spin from "../public/assets/icons/Spin.svg";

export const hexTorgba = (color, alpha) => {
  if (alpha === undefined) alpha = 1;
  const r = parseInt(color.substr(1, 2), 16);
  const g = parseInt(color.substr(3, 2), 16);
  const b = parseInt(color.substr(5, 2), 16);
  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
};

export const createGradientbackground = (from, to) => {
  return `linear-gradient(to right bottom, ${from},${to})`;
};

export const createGradientText = (from, to) => {
  return `linear-gradient(to right bottom,${hexTorgba(from, 0.85)}, 
  ${hexTorgba(to, 0.85)})`;
};

export const showToast = (mode, message, onClose) => {
  toast.dismiss();
  toast[mode](message, {
    autoClose: 5000,
    theme: "colored",
    // onOpen: (props) => console.log("i opened"),
    onClose: onClose,
    style: {
      fontFamily: "inherit",
    },
    position: toast.POSITION.BOTTOM_CENTER,
  });
};

export const showLoadingToast = (message) => {
  toast.dismiss();
  toast.info(message, {
    autoClose: false,
    style: {
      fontFamily: "inherit",
      boxShadow: "0 1px 10px 0 rgb(0 0 0 / 30%), 0 2px 15px 0 rgb(0 0 0 / 25%)",
      color: "#333",
    },
    position: toast.POSITION.BOTTOM_CENTER,
    icon: <Spin style={{ transform: "scale(1.5)", fill: "#666" }} />,
  });
};

export const objectDateToString = (objDate) => {
  return `${objDate.year}/${objDate.month < 10 ? "0" : ""}${objDate.month}/${
    objDate.day < 10 ? "0" : ""
  }${objDate.day}`;
};

// export const stringDateToObject = (stringDate) => {
//   const arrDate = stringDate.split("/");
//   return { year: arrDate[0], month: arrDate[1], day: arrDate[2] };
// };
