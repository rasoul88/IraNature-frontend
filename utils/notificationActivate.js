import { post } from "axios";

const urlBase64ToUint8Array = (base64String) => {
  var padding = "=".repeat((4 - (base64String.length % 4)) % 4);
  var base64 = (base64String + padding).replace(/\-/g, "+").replace(/_/g, "/");

  var rawData = window.atob(base64);
  var outputArray = new Uint8Array(rawData.length);

  for (var i = 0; i < rawData.length; ++i) {
    outputArray[i] = rawData.charCodeAt(i);
  }
  return outputArray;
};

const configurePushSub = () => {
  if (!("serviceWorker" in navigator)) {
    return;
  }

  var reg;
  navigator.serviceWorker.ready
    .then(function (swreg) {
      reg = swreg;
      return swreg.pushManager.getSubscription();
    })
    .then(function (sub) {
      if (sub === null) {
        // Create a new subscription
        var vapidPublicKey =
          "BKMOXd1ARsXXz97HxAFb6tU2BmVdhDJkvpscv77UWadgtLidzKZUxHMloFHdOx18kuYxHkYUTHGzbW0I_cwwAyE";
        var convertedVapidPublicKey = urlBase64ToUint8Array(vapidPublicKey);
        return reg.pushManager.subscribe({
          userVisibleOnly: true,
          applicationServerKey: convertedVapidPublicKey,
        });
      } else {
      }
    })
    .then((newSub) => {
      if (newSub) return post("/subscriptions", newSub);
    })
    .then((res) => {
      console.log("new subscription", res);
    })
    .catch(function (err) {
      console.log(err);
    });
};

export const askForNotificationPermission = () => {
  Notification.requestPermission((result) => {
    if (result !== "granted") {
    } else {
      configurePushSub();
    }
  });
};
