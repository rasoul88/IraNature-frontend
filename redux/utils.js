import { objectDateToString } from "../utils/functions";
export const createQuery = (filterItems, dataLimits) => {
  let query = "";
  for (let item in filterItems) {
    if (item === "name" && filterItems[item].length > 0)
      query += `name[regex]=${filterItems[item]}&`;

    if (item === "days") {
      if (filterItems[item][0] > dataLimits[item].min)
        query += `duration[gte]=${filterItems[item][0]}&`;
      if (filterItems[item][1] < dataLimits[item].max)
        query += `duration[lte]=${filterItems[item][1]}&`;
    }

    if (item === "price") {
      if (filterItems[item][0] > dataLimits[item].min)
        query += `price[gte]=${filterItems[item][0]}&`;
      if (filterItems[item][1] < dataLimits[item].max)
        query += `price[lte]=${filterItems[item][1]}&`;
    }

    if (
      item === "dateRange" &&
      filterItems[item].from &&
      filterItems[item].to
    ) {
      query += `startDate[gte]=${objectDateToString(filterItems[item].from)}&`;
      query += `startDate[lte]=${objectDateToString(filterItems[item].to)}&`;
    }

    if (item === "startLocation" && filterItems[item] && filterItems[item][0]) {
      query += `startLocation=${filterItems[item][0].name}&`;
    }

    if (
      item === "difficulty" &&
      (filterItems[item].length === 1 || filterItems[item].length === 2)
    ) {
      filterItems[item].forEach((el) => (query += `difficulty=${el}&`));
    }

    if (
      item === "maxParticipants" &&
      filterItems[item][0] < dataLimits[item].max
    ) {
      query += `maxGroupSize[lte]=${filterItems[item][0]}&`;
    }

    if (item === "page") {
      query += `page=${filterItems[item]}&`;
    }
  }

  query = query.slice(0, query.length - 1);

  return query;
};
