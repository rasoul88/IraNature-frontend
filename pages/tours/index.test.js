import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { findByTestAttribute } from "../../test/utils";
import ToursPage from "./index";

const setup = (props = {}) => {
  return shallow(<ToursPage {...props} />);
};

test("renders ToursPage correctly", () => {
  const wrapper = setup();
  const component = findByTestAttribute(wrapper, "component-tours-page");
  expect(component.length).toBe(1);
  expect(toJson(wrapper)).toMatchSnapshot();
});
