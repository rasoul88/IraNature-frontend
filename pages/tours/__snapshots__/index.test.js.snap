// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ToursPage correctly 1`] = `
<indexstyles__PageContainer
  data-test="component-tours-page"
>
  <Connect(UnconnectedFilterPanel) />
  <indexstyles__ContentContainer>
    <SecodaryHeading>
       تورهای فعال
    </SecodaryHeading>
    <indexstyles__CardsContainer />
    <indexstyles__PaginateContainer>
      <Paginate
        data-test="tours-pagination"
        maxPage={18}
        onChange={[Function]}
        selectedPage={1}
      />
    </indexstyles__PaginateContainer>
  </indexstyles__ContentContainer>
</indexstyles__PageContainer>
`;
